core/lib/zksync_core/src/state_keeper/metrics.rs

```diff
@@ -0,0 +1,378 @@
+    /// Latency of sealing a miniblock split by the stage.
+    #[metrics(buckets = Buckets::LATENCIES)]
+    sealed_time_stage: Family<MiniblockSealLabels, Histogram<Duration>>,
+    /// Number of entities stored in Postgres during a specific stage of sealing a miniblock.
+    #[metrics(buckets = COUNT_BUCKETS)]
+    sealed_entity_count: Family<MiniblockSealLabels, Histogram<usize>>,
+    /// Latency of sealing a miniblock split by the stage and divided by the number of entiries
+    /// stored in the stage.
+    #[metrics(buckets = Buckets::LATENCIES)]
+    sealed_entity_per_unit: Family<MiniblockSealLabels, Histogram<Duration>>,
+}
+
+impl MiniblockMetrics {
+    pub(super) fn start(&self, stage: MiniblockSealStage, is_fictive: bool) -> SealProgress<'_> {
+        let labels = MiniblockSealLabels {
+            stage,
+            is_fictive: if is_fictive { "true" } else { "false" },
+        };
+        SealProgress {
+            target: "Miniblock",
+            stage_name: format!("{stage:?}"),
+            latency: self.sealed_time_stage[&labels].start(),
+            entity_count: &self.sealed_entity_count[&labels],
+            latency_per_unit: &self.sealed_entity_per_unit[&labels],
+        }
+    }
+}
+
+#[vise::register]
+pub(super) static MINIBLOCK_METRICS: vise::Global<MiniblockMetrics> = vise::Global::new();
+
+/// Tracking progress of L1 batch or miniblock sealing.
+#[must_use = "Progress must be `observe()`d"]
+#[derive(Debug)]
+pub(super) struct SealProgress<'a> {
+    target: &'static str,
+    stage_name: String,
+    latency: LatencyObserver<'a>,
+    entity_count: &'a Histogram<usize>,
+    latency_per_unit: &'a Histogram<Duration>,
+}
+
+impl SealProgress<'_> {
+    pub fn observe(self, count: impl Into<Option<usize>>) {
+        const MIN_STAGE_DURATION_TO_REPORT: Duration = Duration::from_millis(10);
+
+        let count = count.into();
+        let elapsed = self.latency.observe();
+        if elapsed > MIN_STAGE_DURATION_TO_REPORT {
+            let target = self.target;
+            let stage = &self.stage_name;
+            tracing::debug!(
+                "{target} execution stage {stage} took {elapsed:?} with count {count:?}"
+            );
+        }
+
+        if let Some(count) = count {
+            self.entity_count.observe(count);
+            if count > 0 {
+                self.latency_per_unit.observe(elapsed.div_f64(count as f64));
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "command", rename_all = "snake_case")]
+pub(super) enum ExecutorCommand {
+    ExecuteTx,
+    StartNextMiniblock,
+    RollbackLastTx,
+    FinishBatch,
+}
+
+const GAS_PER_NANOSECOND_BUCKETS: Buckets = Buckets::values(&[
+    0.01, 0.03, 0.1, 0.3, 0.5, 0.75, 1., 1.5, 3., 5., 10., 20., 50.,
+]);
+
+/// Executor-related state keeper metrics.
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "state_keeper")]
```

Comment:
Note the discrepancy between this prefix and the prefix for `StateKeeperMetrics` (`server_state_keeper`). IMO, it would make sense to use the same prefix in both cases.